lib/spack/spack/extensions.py

```diff
@@ -11,10 +11,48 @@
 import types
 
 import llnl.util.lang
-import llnl.util.tty as tty
+import spack.cmd
 import spack.config
+from spack.error import SpackError
 
-extension_regexp = re.compile(r'spack-([\w]*)')
+_extension_regexp = re.compile(r'spack-([\w]*)')
+_extension_command_map = None
+
+
+def _init_extension_command_map():
+    """Return the list of paths where to search for command files."""
+    global _extension_command_map
+    if _extension_command_map is None:
+        _extension_command_map = {}
+        extension_paths = spack.config.get('config:extensions') or []
+        for path in extension_paths:
+            extension = extension_name(path)
+            command_path = os.path.join(path, extension, 'cmd')
+            commands = spack.cmd.find_commands(command_path)
+            _extension_command_map.update(
+                ((command, path) for command in
+                 commands if command not in _extension_command_map))
+
+
+def reset_command_cache():
```

Comment:
Wouldn't this be better as a fixture? If we are changing a global just for test purposes it's probably better to use `monkeypatch` on the module attribute where appropriate.

Reply:
I was unfamiliar with `monkeypatch` until very recently. If you can't see a non-test use case for `reset_command_cache`, I can do as you suggest.